src/protocols/aeternity/composables/aeNames.ts

```diff
@@ -0,0 +1,290 @@
+import {
+  ref,
+  watch,
+} from 'vue';
+import {
+  fetchAllPages,
+  fetchJson,
+  handleUnknownError,
+} from '@/utils';
+import { useMiddleware } from '@/composables/middleware';
+import { useAccounts } from '@/composables/accounts';
+import { Encoded } from '@aeternity/aepp-sdk';
+import { useTransactionList } from '@/composables/transactionList';
+import type {
+  ChainName,
+  IDefaultComposableOptions,
+  IName,
+  ITransaction,
+} from '@/types';
+import {
+  AUTO_EXTEND_NAME_BLOCKS_INTERVAL,
+  PROTOCOL_AETERNITY,
+  UPDATE_POINTER_ACTION,
+} from '@/constants';
+import { useAeSdk } from '@/composables/aeSdk';
+import { isInsufficientBalanceError } from '@/protocols/aeternity/helpers';
+import {
+  IAddressNamePair,
+  IAuction,
+  IAuctionBid,
+} from '@/types';
+import { useAeNetworkSettings } from '@/protocols/aeternity/composables/aeNetworkSettings';
+import { useModals } from '@/composables/modals';
+import { tg } from '@/store/plugins/languages';
+import {
+  createPollingBasedOnMountedComponents,
+  useStorageRef,
+} from '@/composables/composablesHelpers';
+
+const POLLING_INTERVAL = 10000;
+
+interface IUpdatePointerParams {
+  name: ChainName;
+  address?: any;
+  type?: keyof typeof UPDATE_POINTER_ACTION;
+}
+
+interface IAuctionEntryParams {
+  name: ChainName;
+  expiration: number;
+  bids: IAuctionBid[];
+}
+
+const ownedNames = useStorageRef<IName[]>([], 'owned-names');
+const pendingAutoExtendNames = ref<ChainName[]>([]);
+const areNamesFetching = ref(false);
+const defaults = ref<Record<string, ChainName>>({});
```

Comment:
To make i more readable can you please rename this constant to something like `defaultNamesRegistry`?
Also it would be nice if you could add a comment about what the `string` stands for.

Reply:
I see that this `string` is a `<NodeNetworkID>-<address>`. Maybe we could make the object multidimentional instead?
So the structure would look as follows:
```
{
  'mainnet': {
    'ak_abcdfg...': 'somename.chain',
  }
}
```